### ISSUE TITLE
Group messages are displayed in received order, not sent order

### ISSUE THREAD

**Issue by uselpa** [signalapp/Signal-iOS#2983]
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
Group messages are displayed in received order, not sent order. Messages are not received in sent order so the conversation is hard to follow if the receiving phone has been offline during that time.

### Steps to reproduce
- put receiving phone offline for 1 h
- send a reasonable amount of messages (ideally from several other phones)
- put receiving phone online

**Actual result:** 
Messages are out of order and conversation threat is hard to follow.

**Expected result:** 
Messages should be ordered by sent time.

### Device info
**Device**: iPhone 8 GSM
**iOS version**: 11.2.5 (15D60)
**Signal version:** 2.19.5.0

### Link to debug log
sent to support@whispersystems.org yesterday

**Comment by michaelkirk** [comment]
This is as intended.

If we ordered them by "sent time", that means we'd be potentially inserting messages back in your message history. If it happened far enough back, it would be easy to miss messages.

So admittedly, the current state leaves potential for confusion, but we've considered behavior like your proposal behavior and decided that would be worse.

**Comment by uselpa** [comment]
Why not add a "reorder" button to the conversation view then? This way messages won't be missed and the ordering step is optional.

**Comment by uselpa** [comment]
Ideally the button would only show up if the thread is unordered.

**Comment by uselpa** [comment]
any chance to get a feedback?
